File: backend/src/main/java/com/hgw/gestionale/brand/mapper/BrandMapper.java
package com.hgw.gestionale.brand.mapper;

import com.hgw.gestionale.brand.dto.BrandResponse;
import com.hgw.gestionale.brand.dto.CreateBrandRequest;
import com.hgw.gestionale.brand.dto.UpdateBrandRequest;
import com.hgw.gestionale.brand.entity.Brand;

public final class BrandMapper {
    private BrandMapper() {}

    public static Brand toEntity(CreateBrandRequest request) {
        return Brand.builder()
                .nome(request.nome())
                .build();
    }

    public static BrandResponse toResponse(Brand brand) {
        return new BrandResponse(
                brand.getId(),
                brand.getNome()
        );
    }

    public static void updateEntity(Brand brand, UpdateBrandRequest request) {
        brand.setNome(request.nome());
    }
}

File: backend-node/src/mapper/brand.mapper.ts
import { BrandResponse, UpdateBrandRequest } from "../dto/brand.dto";
import { brands } from "../schema/brands";

type BrandEntity = typeof brands.$inferSelect;

export class BrandMapper {

    static toResponse(entity: BrandEntity): BrandResponse {
        return {
            id: entity.id,
            nome: entity.nome,
        };
    }

    static updateEntity(entity: BrandEntity, request: UpdateBrandRequest): void {
        entity.nome = request.nome;
    }

}
